homeassistant/helpers/dispatcher.py

```diff
@@ -34,7 +34,7 @@ def dispatcher_connect(
 
     def remove_dispatcher() -> None:
         """Remove signal listener."""
-        run_callback_threadsafe(hass.loop, async_unsub).result()
+        hass.loop.call_soon_threadsafe(async_unsub)
```

Comment:
I think this should be safe as I don't think the sync code needs to wait for the event loop to run.

This is almost unused in the codebase as nearly everything is async at this point. The only calls I could find are
```
homeassistant/components/asterisk_mbox/__init__.py:        dispatcher_connect(self.hass, SIGNAL_MESSAGE_REQUEST, self._request_messages)
homeassistant/components/asterisk_mbox/__init__.py:        dispatcher_connect(self.hass, SIGNAL_CDR_REQUEST, self._request_cdr)
homeassistant/components/asterisk_mbox/__init__.py:        dispatcher_connect(self.hass, SIGNAL_DISCOVER_PLATFORM, self._discover_platform)

```

Reply:
I think this is unexpected, that the listener is still registered after this call returns. I don't think we should do this.

Reply:
I think thats the right call. In this case this function is almost never called so the benefit to reducing the wait time likely isn't worth the change risk.